conda/resolve.py

```diff
@@ -671,15 +671,13 @@ def generate_feature_count(self, C, trackers):
 
     def generate_feature_metric(self, C, groups, specs):
         eq = {}
+        total = 0
         for name, group in iteritems(groups):
             nf = [len(self.features(fn)) for fn in group]
             maxf = max(nf)
```

Comment:
If we're worried about performance here, why not turn `nf` into a generator.

    maxf = max((len(self.feature(fn) for fn in group)))